### ISSUE TITLE
When sample packing is enabled with Gemma 7b, I experience high loss.

### ISSUE THREAD

**Issue by hcsolakoglu** [huggingface/trl#1392]
Loss graphs provided below. Same data same settings. Only diffrence is packing.

![image](https://github.com/huggingface/trl/assets/155680432/f1bf0216-332a-4b71-ab9a-da2a2d48de27)
![image](https://github.com/huggingface/trl/assets/155680432/6952b703-cd82-4ada-aba0-26c47259b724)

Name: torch
Version: 2.1.0+cu121
Name: transformers #Don't let the Transformers version fool you; the Gemma rope fix has been applied.
Version: 4.38.1
Name: datasets
Version: 2.18.0
Name: accelerate
Version: 0.27.2
Name: peft
Version: 0.9.0
Name: trl
Version: 0.7.11
Name: bitsandbytes
Version: 0.42.0
V12.2.140

**Comment by younesbelkada** [comment]
Hi @hcsolakoglu 
I suspect this is related to the BOS token as Gemma seems quite sensible to that token, can you confirm in both scenarios the BOS token is set correctly in the encoded sentence?

**Comment by hcsolakoglu** [comment]
> Hi @hcsolakoglu I suspect this is related to the BOS token as Gemma seems quite sensible to that token, can you confirm in both scenarios the BOS token is set correctly in the encoded sentence?

I did more tests. My observations are as follows: This problem does not occur with multipacking in Axolotl, only a high loss value occurs when I use TRL's packing with sfttrainer. So model is not the problem. In these tests, I saw that the high loss decreased very slowly, but I realized that the initial high loss affected the training quality of the model. I think bos token was added automatically when the dataset and related fields were given to the sfttrainer. In my initial test i was adding only eos manually. Upon your warning, I added bos token manually too. The results did not change. It was probably added automaticly in my first tests.
